### DISCUSSION TITLE
How to change the size of font?

### DISCUSSION THREAD

**Discussion by yao5461** [posquit0/Awesome-CV#464] [Q&A]
Hi~
I want to make the font size smaller, How?
Thanks.

**Comment by johannesbottcher** [comment]
`\documentclass[11pt, a4paper]{awesome-cv}` ... change `11` to `10`.

**Comment by yao5461** [comment]
Thanks! 
And how to reduce space between sections?
I'm a newbie. Thanks!

**Comment by johannesbottcher** [comment]
`\renewcommand{\acvSectionTopSkip}{0mm}`

**Comment by arekolek** [comment]
@johannesbottcher Does it really change the font size? In my case the font stays the same regardless of whether I choose 10, 11 or 12pt. Only line spacing is affected.

Font sizes seem to be fixed because of how the styles are defined in [awesome-cv.cls](https://github.com/posquit0/Awesome-CV/blob/67b2708a9e33c205a30d49e6973661459539418d/awesome-cv.cls#L226), for example:

```
\newcommand*{\paragraphstyle}{\fontsize{9pt}{1em}\bodyfontlight\upshape\color{text}}
```

Changing the font option to `\documentclass` can't affect `cvparagraph` environment font size because it is fixed to `\fontsize{9pt}{1em}`.

It can't change any other font size, for the same reason.

**Comment by johannesbottcher** [comment]
@arekolek For simple text, yes.

**Comment by arekolek** [comment]
@johannesbottcher Yet the examples never use any simple text.

**Comment by johannesbottcher** [comment]
@arekolek If i change `11pt` to `10pt` in the example CV, i can see more stuff on the page. `cvitems` uses normal font without any changers. 

More stuff on the page is not a real reason to claim the font size has changed. I didn't do a real check. Might just a well be the different baselineskip when no special environment is in charge. If it really is like that, the `11pt` needs to be removed from the example because it is misleading. As i said, i never checked. CV-templates aren't any good.

**Comment by arekolek** [comment]
@johannesbottcher Actually, `\cvitems` is wrapped inside `\descriptionstyle` in the examples.

I'm guessing @posquit0  put the `11pt` in there to get the line spacing, since it doesn't affect the font size at all.

**Comment by JanHendrikDolling** [comment]
checkout my fork at https://github.com/JanHendrikDolling/Awesome-CV
You could change the font size and line stretch with the following commands.
FYI: I'd only used the cv template.

% font size for different elements
\sizesection{16pt}
\sizesubsection{12pt}
\sizeentrytitle{10pt}
\sizeentryposition{8pt}
\sizeentrydate{8pt}
\sizeentrylocation{9pt}
\sizedescription{9pt}

% line stretch
\stretchdescription{1.2}

**Comment by haowenz** [comment]
@arekolek 
Hi,
I tried to change the font size here but the font size didn't change. 
`\newcommand*{\paragraphstyle}{\fontsize{9pt}{1em}\bodyfontlight\upshape\color{text}}`

Do you know how to change the front size?

Thanks.

**Comment by johannesbottcher** [comment]
This will give you an error because `\paragraphstyle` is already defined. The `cvparagraph` environment isn't used in the example templates. 

There are about 30 styles that can be changed individually. There is no simple switch to make fonts bigger. 

That could be done, but it would require many (*many*) lines of code.

**Comment by fguardini** [comment]
Coming back to this as I was looking for the same issue, all the settings to be changed are in .cls, as I guess @yao5461 was looking for this, under the "Configuration for Styles - Configure styles for each element" section. Most likely he needed the bullet points configuration, which can be found on the line

`\newcommand*{\descriptionstyle}[1]{{\fontsize{9pt}{1em}\bodyfontlight\upshape\color{text} #1}}`

where 9pt is the size, knowing that there are 34 different text elements that should be modified accordingly :P 

Out of curiosity, wouldn't it be possible to have a dynamic reference for font size? For example in the document the smallest font size is 8pt, if that is defined as x and every other font is defined as x, x+1, x+2 etc just changing the small font size would update everything else...

**Comment by johannesbottcher** [comment]
> Out of curiosity, wouldn't it be possible to have a dynamic reference for font size? For example in the document the smallest font size is 8pt, if that is defined as x and every other font is defined as x, x+1, x+2 etc just changing the small font size would update everything else...

That is the default LaTeX approach, This template handles stuff a bit differently. As i said, many lines of code would need to be changed to allow different font sizes by default. Standard book class provides three different font sizes to choose from, each one needs an extra file of about 200 lines of code. 

For a CV, a bit less would be needed. On the other hand, a templates purpose is to set the style and layout. Though a user is able to change some stuff, not everything is possible.

**Comment by codemongkey** [comment]
so ,how to change the size of font

**Comment by rhecht95** [comment]
> checkout my fork at https://github.com/JanHendrikDolling/Awesome-CV You could change the font size and line stretch with the following commands. FYI: I'd only used the cv template.
> 
> % font size for different elements \sizesection{16pt} \sizesubsection{12pt} \sizeentrytitle{10pt} \sizeentryposition{8pt} \sizeentrydate{8pt} \sizeentrylocation{9pt} \sizedescription{9pt}
> 
> % line stretch \stretchdescription{1.2}

@JanHendrikDolling where does the \stretchdescription command get placed to take effect? I'm having trouble making this work but I also dont know too much about LaTex syntax. My best guess was somewhere on line 210:

\newcommand*{\descriptionstyle}[1]{{\fontsize{9pt}{1em}\bodyfontlight\upshape\color{text} #1}}

**Comment by JanHendrikDolling** [comment]
I haven't used the template for a long time. [Try the following line](https://github.com/JanHendrikDolling/Awesome-CV/blob/0d028f2a13cdbcf40e5b700d64690ebd898e9157/examples/cv.tex#L61).

**Reply by rhecht95** [reply]
Thanks! I actually managed it a different way with \linesep in the code block at line 684:

% Define an environment for cvitems(for cventry)
\newenvironment{cvitems}{%
  \vspace{-4.0mm}
  \begin{justify}
  \begin{itemize}[leftmargin=2ex, nosep, noitemsep]
    \setlength{\parskip}{0pt}
    \setlength{\itemsep}{0.6em}
    \renewcommand{\labelitemi}{\bullet}
}{%
  \end{itemize}
  \end{justify}
  \vspace{-4.0mm}
}
